neverbleed.c

```diff
@@ -635,13 +687,25 @@ static EC_KEY *daemon_get_ecdsa(size_t key_index)
 
 static size_t daemon_set_ecdsa(EC_KEY *ec_key)
 {
-    size_t index;
-
     pthread_mutex_lock(&daemon_vars.keys.lock);
-    if ((daemon_vars.keys.ecdsa_keys = realloc(daemon_vars.keys.ecdsa_keys,
-                                               sizeof(*daemon_vars.keys.ecdsa_keys) * (daemon_vars.keys.ecdsa_size + 1))) == NULL)
-        dief("no memory");
-    index = daemon_vars.keys.ecdsa_size++;
+    if (!daemon_vars.keys.reserved_ecdsa_size || (daemon_vars.keys.ecdsa_size >= daemon_vars.keys.reserved_ecdsa_size)) {
+        size_t size = new_key_array_size(daemon_vars.keys.reserved_ecdsa_size);
+        if ((daemon_vars.keys.keys = realloc(daemon_vars.keys.ecdsa_keys, sizeof(*daemon_vars.keys.ecdsa_keys) * size)) == NULL)
+            dief("no memory");
+
+        daemon_vars.keys.bita_ecdsa_keys = bita_realloc(daemon_vars.keys.bita_ecdsa_keys, daemon_vars.keys.reserved_ecdsa_size, size);
+        daemon_vars.keys.reserved_ecdsa_size = size;
```

Comment:
Sure! wasn't happy about the repetition either but was trying to avoid wrapping the keys. Perhaps  can wrap the sizes/bit-array stuff instead. Will improve this code, thanks!